### PR TITLE
Add hostname option

### PR DESCRIPTION
**Pull request by jared-w**
Set the PR into this branch because that's what saw-script is currently pulling in as a submodule.

### CODE REVIEW

File: argo/src/Argo/DefaultMain.hs [hunk 1]
```diff
@@ -166,18 +173,20 @@ realMain theApp opts =
   case transportOpt opts of
     StdIONetstring ->
       serveStdIONS theApp
-    SocketNetstring sessionOpt publicityOpt portOpt ->
+    SocketNetstring sessionOpt publicityOpt portOpt hostNameOpt ->
       do sessionResult <- getOrLockSession sessionOpt publicityOpt portOpt
+         let hostname = case hostNameOpt of
```
Comment (david-christiansen) [inline argo/src/Argo/DefaultMain.hs:178]: As a matter of style, I very weakly prefer `fromMaybe` for this pattern. But it's not a big deal.

**Review by david-christiansen — COMMENTED** [review]
This looks good to me.

@atomb, do you think this should go onto the designated branch, or into `master`? I don't have a lot of insight into how you're working through the server code transition, or what would be most convenient there.

File: argo/src/Argo/DefaultMain.hs [hunk 2]
```diff
@@ -77,6 +77,13 @@ transport =
        Opt.help "Create or look up a globally available session")
       <|> pure Nothing
 
+    hostname =
+      (fmap Just . Opt.strOption $
+        Opt.long "host"
+        <> Opt.metavar "HOST"
+        <> Opt.help "Make server available at the secified hostname")
```
Comment (kquick) [inline argo/src/Argo/DefaultMain.hs:84]: specified

**Comment by david-christiansen** [comment]
Now that #93 and #92 are merged, I do think this should target `master` instead.

**Comment by jared-w** [comment]
I updated it to target master instead.

**Comment by david-christiansen** [comment]
Thank you!

**Pull request merged**